### DISCUSSION TITLE
Using a tasmota device to monitor a non tasmota device and reboot it on wifi loss

### DISCUSSION THREAD

**Discussion by baskarz** [arendst/Tasmota#22498] [Q&A] [answered]
Hi all.
Wondering if it's possible for a tasmotised esp relay to monitor a non esp device on my wifi network and if it drops off the network use the esp relay to power cycle the non tasmota device?
The device in question is an esp32 that runs a custom firmware a friend made that monitors my house battery voltage and shuts off solar if battery voltage exceeds a high set point. 

I was thinking of maybe a rule that could monitor the esp32 ip and if it disappears then cycle the relay to power cycle the esp32.
Just not too sure where to start.

My friend is working on a hardware watchdog to solve this issue but it won't be ready for a few months due to work commitments.  So this would be just a temporary solution

**Comment by sfromis** [comment, marked as answer]
Depending on how the other device acts on the network, you could periodically try a `Ping` or `WebQuery` to contact it. Example rules:
https://tasmota.github.io/docs/Rules/#watchdog-for-wi-fi-router-or-modem

If the Tasmota device is ESP32-based, you could also use Berry for other types of network access as a similar heartbeat check.

**Reply by baskarz** [reply]
Thanks sfromis!
That's exactly what I'm looking for.
Much appreciated
